test/tests/functional/pbs_ralter.py

```diff
@@ -2234,3 +2235,32 @@ def test_select_fail_revert(self):
         a = {'Resource_List.select': '8:ncpus=1', 'Resource_List.ncpus': 8,
              'Resource_List.nodect': 8}
         self.server.expect(RESV, a, id=rid)
+
+    def test_resv_resc_assigned(self):
+        """
+        Test that when an ralter -D is issued, the resources on the node
+        are still correct
+        """
+
+        offset = 60
+        dur = 60
+        select = "4:ncpus=1"
+
+        rid, start, end = \
+            self.submit_and_confirm_reservation(offset, dur, select=select)
+
+        self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+
+        sleepdur = start - time.time()
+        self.logger.info('Sleeping until reservation starts')
+        self.server.expect(RESV,
+                           {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')},
+                           offset=sleepdur)
+
+        self.server.alterresv(rid, {ATTR_resv_duration: '10:00'})
```

Comment:
can we reduce the duration instead of increasing it? we can then wait to check that the reservation does not exist when the duration is reduced.

Reply:
im sorry arun, i don't really understand
the length of the duration doesn't really matter, it just needs to change. we don't ahve to wait for the duration.

Reply:
Sorry about being unclear. What I mean to say was that the bug description states - "_If pbs_ralter -D would change the end time to a later time, the reservation would still end at the original time._" So would it not be useful to have a test that captures the modified reservation actually ends at the modified time.

The reason I suggested reducing the duration was that I thought it does not matter if duration is extended or reduced, in both the cases (because of the bug) the reservation would end at the original end time.

Reply:
There is a test that decreases the duration, then checks to see if the reservation is ended, so I think that's proof enough for the reservation's duration being reduced.
There is a test that increases the duration, I make sure that the test is still running after the original end time has passed. That should be enough proof for the reservation's duration being increased.

This last test is for the resources, to make sure they're not being double counted.

Reply:
ah ok, if it already exists then fine.
Thanks!